### PR TITLE
add peft to recipe qwen3vl

### PR DESCRIPTION
**Pull request by yashaswikarnati**
# What does this PR do ?

Add PEFT to qwen 3VL recipe
image

# Changelog

- Add specific line by line info of high level changes in this PR.

# GitHub Actions CI

See the [CI section](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md#-running-github-ci)in the Contributing doc for how to trigger the CI. A Nvidia developer will need to approve and trigger the CI for external contributors.

# Before your PR is "Ready for review"

**Pre checks**:

- [ ] Make sure you read and followed [Contributor guidelines](https://github.com/NVIDIA-NeMo/Megatron-Bridge/blob/main/CONTRIBUTING.md)
- [ ] Did you write any new necessary tests?
- [ ] Did you add or update any necessary documentation?
- [ ] Does the PR affect components that are optional to install? (Ex: Numba, Pynini, Apex etc)
  - [ ] Reviewer: Does the PR have correct import guards for all optional libraries?

If you haven't finished some of the above items you can still open "Draft" PR.

# Additional Information

- Related to # (issue)

## Release Notes

* **Documentation**
  * Added PEFT (Parameter-Efficient Fine-Tuning) section to Qwen3-VL documentation with LoRA and DoRA examples for multiple model sizes.

* **New Features**
  * Enabled dynamic PEFT configuration support for Qwen3-VL finetuning via CLI overrides.
  * Extended configuration API with new Qwen3-VL model variants (8B, 30B, 235B).
  * Added customizable learning rate configuration for finetuning workflows.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

**Review by cuichenx — DISMISSED** [review]
LGTM
